webhook/workspace/mutating_cfg.go

```diff
@@ -23,54 +23,56 @@ import (
 const (
 	MutateWebhookCfgName       = "controller.devfile.io"
 	mutateWebhookPath          = "/mutate"
-	mutateWebhookFailurePolicy = v1beta1.Fail
+	mutateWebhookFailurePolicy = admregv1.Fail
 )
 
 // BuildMutateWebhookCfg creates the mutating webhook configuration for the controller
-func BuildMutateWebhookCfg(namespace string) *v1beta1.MutatingWebhookConfiguration {
+func BuildMutateWebhookCfg(namespace string) *admregv1.MutatingWebhookConfiguration {
 	mutateWebhookFailurePolicy := mutateWebhookFailurePolicy
 	mutateWebhookPath := mutateWebhookPath
 	labelExistsOp := metav1.LabelSelectorOpExists
-	equivalentMatchPolicy := v1beta1.Equivalent
-	sideEffectsNone := v1beta1.SideEffectClassNone
-	webhookClientConfig := v1beta1.WebhookClientConfig{
-		Service: &v1beta1.ServiceReference{
+	equivalentMatchPolicy := admregv1.Equivalent
+	sideEffectsNone := admregv1.SideEffectClassNone
+	webhookClientConfig := admregv1.WebhookClientConfig{
+		Service: &admregv1.ServiceReference{
 			Name:      server.WebhookServerServiceName,
 			Namespace: namespace,
 			Path:      &mutateWebhookPath,
 		},
 		CABundle: server.CABundle,
 	}
 
-	workspaceMutateWebhook := v1beta1.MutatingWebhook{
+	workspaceMutateWebhook := admregv1.MutatingWebhook{
 		Name:          "mutate.devworkspace-controller.svc",
 		FailurePolicy: &mutateWebhookFailurePolicy,
 		ClientConfig:  webhookClientConfig,
 		SideEffects:   &sideEffectsNone,
-		Rules: []v1beta1.RuleWithOperations{
+		Rules: []admregv1.RuleWithOperations{
 			{
-				Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
-				Rule: v1beta1.Rule{
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+				Rule: admregv1.Rule{
 					APIGroups:   []string{"workspace.devfile.io"},
 					APIVersions: []string{"v1alpha1", "v1alpha2"},
 					Resources:   []string{"devworkspaces"},
 				},
 			},
 			{
-				Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
-				Rule: v1beta1.Rule{
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+				Rule: admregv1.Rule{
 					APIGroups:   []string{"controller.devfile.io"},
 					APIVersions: []string{"v1alpha1"},
 					Resources:   []string{"devworkspaceroutings", "components"},
 				},
 			},
 		},
+		AdmissionReviewVersions: []string{"v1beta1", "v1"},
```

Comment:
why it's needed? when should we remove it?

Reply:
IIRC On OpenShift 4.9 it failed without it. I'll double check again though

Reply:
OK, seems I reproduced the issue you talk about on OpenShift 4.9
> Error from server (InternalError): error when creating "./samples/flattened_theia-next.yaml": Internal error occurred: failed calling webhook "mutate.devworkspace-controller.svc": received invalid webhook response: expected response.uid="07b5b66c-0d84-48bb-939f-c4f6bdcaf3a5", got ""

I assume we should be able to remove it after we upgrade controller-runtime in Angel's PR.